gui/src/app/pages/HomePage/LoadProjectWindow.tsx

```diff
@@ -24,63 +23,64 @@ const LoadProjectWindow: FunctionComponent<LoadProjectWindowProps> = ({
+          files: fileManifest,
+          clearExisting: true,
+        });
+        onClose();
+      } catch (e: any) {
+        setErrorText(e.message);
+      }
+    },
+    [onClose, update],
+  );
 
   const importUploadedFiles = useCallback(
     async (o: { replaceProject: boolean }) => {
       const { replaceProject } = o;
       if (!filesUploaded) return;
       try {
-        if (
-          filesUploaded.length === 1 &&
-          filesUploaded[0].name.endsWith(".zip")
-        ) {
-          // a single .zip file
-          const fileManifest = await deserializeZipToFiles(
-            filesUploaded[0].content,
-          );
-          update({
-            type: "loadFiles",
-            files: fileManifest,
-            clearExisting: replaceProject,
-          });
-        } else if (
-          filesUploaded.length === 1 &&
-          filesUploaded[0].name.endsWith(".stan")
-        ) {
-          // a single .stan file
-          if (replaceProject) {
-            update({ type: "retitle", title: filesUploaded[0].name });
-          }
-          const fileManifest: Partial<FieldsContentsMap> = {
-            stanFileContent: parseFile(filesUploaded[0].content),
-          };
-          update({
-            type: "loadFiles",
-            files: fileManifest,
-            clearExisting: replaceProject,
-          });
-        } else {
-          const files: Partial<FileRegistry> = {};
-          for (const file of filesUploaded) {
-            if (!Object.values(FileNames).includes(file.name as any)) {
-              throw Error(`Unrecognized file: ${file.name}`);
+        let stanFileName = "";
+        const files: Partial<FileRegistry> = {};
+
+        for (const file of filesUploaded) {
+          if (file.name.endsWith(".stan")) {
+            if (stanFileName !== "") {
+              throw Error("Only one .stan file can be uploaded at a time");
             }
-            files[file.name as FileNames] = parseFile(file.content);
+            files["main.stan"] = parseFile(file.content);
+            stanFileName = file.name;
+            continue;
+          }
+          if (file.name.endsWith(".zip")) {
+            throw Error(".zip files cannot be uploaded alongside other files");
           }
 
-          const fileManifest = mapFileContentsToModel(files);
-          update({
-            type: "loadFiles",
-            files: fileManifest,
-            clearExisting: replaceProject,
-          });
+          if (!Object.values(FileNames).includes(file.name as any)) {
+            throw Error(`Unsupported file name: ${file.name}`);
+          }
+          files[file.name as FileNames] = parseFile(file.content);
+        }
+
+        const fileManifest = mapFileContentsToModel(files);
+        update({
+          type: "loadFiles",
+          files: fileManifest,
+          clearExisting: replaceProject,
+        });
+
+        if (stanFileName !== "" && fileManifest.meta === undefined) {
```

Comment:
This is supposed to retitle the project if the .stan file uploaded is not called main.stan, right? If so I'm not sure that's working...

Reply:
It appears to be working for me? Though it should probably only do this if you hit 'new project'

Reply:
Ah, it only functions if you don't have a meta.json in the upload. My mistake.

Reply:
Yeah, we let the title in `meta.json` be the definitive one in the case that there are both